materials/views.py

```diff
@@ -1,42 +1,62 @@
-from rest_framework.generics import (CreateAPIView, DestroyAPIView,
-                                     ListAPIView, RetrieveAPIView,
-                                     UpdateAPIView)
+from rest_framework import generics
+from rest_framework.permissions import IsAuthenticated
 from rest_framework.viewsets import ModelViewSet
 
 from materials.models import Course, Lesson
-from materials.serializers import CourseSerializer, LessonSerializer
+from materials.serializers import CourseDetailSerializer, CourseSerializer, LessonSerializer
+from users.permissions import IsModers, IsOwner
 
-"""CRUD для модели курс по Viewsets"""
 
-
-class CourseViewSets(ModelViewSet):
+class CourseViewSet(ModelViewSet):
     queryset = Course.objects.all()
-    serializer_class = CourseSerializer
 
+    def get_permissions(self):
+        if self.action == "create":
+            self.permission_classes = (~IsModers,)
```

Comment:
Для начала нужно проеврить авторизацию
```suggestion
            self.permission_classes = (IsAuthenticated, ~IsModers)
```